### DISCUSSION TITLE
What does "non-existing relay set for server ..." mean?

### DISCUSSION THREAD

**Discussion by mnlipp** [DNSCrypt/dnscrypt-proxy#2208] [Potential issues]
After upgrading to 2.1.2 (OpenWrt) dnscrypt-proxy2 does not work any more. I always get
```
Non-existent relay set for server <some server>
dnscrypt-proxy is waiting for at least one server to be reachable
```
Where `<some server>` is a random pick from my server list. I've looked at the [source](https://github.com/DNSCrypt/dnscrypt-proxy/blob/ca253923eb631ccde49bfcc6cf376ba867485108/dnscrypt-proxy/serversInfo.go#L468) but there's too much unknown context (what are "stamps"?).

I've only used servers from the `relays.md`, but the message also appears if I leave the list empty (reports the problem for a random pick from `relays.md`, I assume).

**Comment by mnlipp** [comment]
Okay, had some more ideas after all.
* I'm quite sure that I had read that you should add all servers used as relays to `disabled_server_names`. This seems to be wrong now. Removing them from the list made dnscrypt attempt a connection (though it still failed)
* `force_tcp = true` finally did the trick.

**Reply by jedisct1** [reply]
It means that you mentioned the name of a relay that doesn't exist.

Maybe a typo.

**Comment by mnlipp** [comment]
Okay, "non-existent relay set" with set as in prsent perfect of set. I read it as "relay set" as in "set of properties". I'm not a native, so I may be wrong about the ambiguity. But I'd suggest to change that in "non-existing relay configuration for ..."

Thanks for the clarification.
